src/daft-plan/src/logical_optimization/rules/push_down_projection.rs

```diff
@@ -230,6 +236,71 @@ impl PushDownProjection {
                 }
             }
             LogicalPlan::ActorPoolProject(upstream_actor_pool_projection) => {
+                // Attempt to merge the current Projection into the upstream ActorPoolProject
+                // if there aren't any actual computations being performed in the Projection, and
+                // if each upstream column is used only once (no common subtrees)
+                if projection
+                    .projection
+                    .iter()
+                    .all(|e| !requires_computation(e))
+                {
+                    let required_column_names = projection
+                        .projection
+                        .iter()
+                        .flat_map(get_required_columns)
+                        .collect_vec();
+                    let distinct_required_column_names =
+                        required_column_names.iter().collect::<IndexSet<_>>().len();
+                    if required_column_names.len() == distinct_required_column_names {
```

Comment:
Very minor comment but a possible way to do the same check that may be more performant is to iterate through the required columns and put them into a set after checking if they already exist in there. That way you can early terminate

Reply:
Thanks, modified!